Frontend/src/components/TeacherPage.jsx

```
@@ -220,18 +227,172 @@ export default function TeacherDashboard() {
+
+      setRecentAnnouncements(recent);
+      setOldAnnouncements(old);
+      setAnnouncements(sortedAnnouncements);
+    } catch (error) {
+      console.error('Error fetching announcements:', error);
+      toast.error('Failed to fetch announcements');
+    }
+  };
+  const fetchAnnouncements = async (classCode) => {
+    try {
+      const token = localStorage.getItem('token');
```

Biome (1.9.4)

[error] Shouldn't redeclare 'fetchAnnouncements'. Consider to delete it or rename it.

'fetchAnnouncements' is defined here:


(lint/suspicious/noRedeclare)